src/plugins/teleport-plugin-react-styled-components/index.ts

```diff
@@ -0,0 +1,69 @@
+import { ComponentPluginFactory, ComponentPlugin } from '../../typings/generators'
+import { generateStyledComponent } from './utils'
+import { traverseElements, transformDynamicStyles } from '../../shared/utils/uidl-utils'
+import { stringToUpperCamelCase } from '../../shared/utils/string-utils'
+import { createJSXSpreadAttribute } from '../../shared/utils/ast-jsx-utils'
+
+interface StyledComponentsConfig {
+  componentChunkName: string
+  importChunkName?: string
+}
+
+export const createPlugin: ComponentPluginFactory<StyledComponentsConfig> = (config) => {
+  const { componentChunkName = 'react-component', importChunkName = 'import-local' } = config || {}
+
+  const reactStyledComponentsPlugin: ComponentPlugin = async (structure) => {
+    const { uidl, chunks, dependencies } = structure
+    const { node } = uidl
+    const componentChunk = chunks.find((chunk) => chunk.name === componentChunkName)
+    if (!componentChunk) {
+      return structure
+    }
+
+    const jsxNodesLookup = componentChunk.meta.nodesLookup
+    const jssStyleMap = {}
+
+    traverseElements(node, (element) => {
+      const { style, key, elementType } = element
+      if (style) {
+        const root = jsxNodesLookup[key]
+        const className = `${stringToUpperCamelCase(key)}Wrapper`
+        let needInjectionOfProps: boolean = false
+        jssStyleMap[className] = transformDynamicStyles(style, (styleValue) => {
+          if (styleValue.content.referenceType === 'prop') {
+            needInjectionOfProps = true
```

Comment:
By default we don't need to spread the props, styled-components will send all the props of the elements down to each styled component. Is there a specific reason for adding this behavior by default? if we need to send custom props to the styled component (ex: from state) we should send them explicitly

Reply:
Hey @alexnm check this out --> https://codesandbox.io/s/q7ywk8647j if i am not spreading props and passing them, the component is not rendering the style, initially, i was not doing this, but once i exported the component to sandbox, i observed this behavior

Reply:
ah, I misread the docs. However, does it make sense to pass all props through, or just the ones which are used in the styled component. I think that in most of the cases you need 1 of the props in the styled component. I rarely see the case where all of them should be passed. Hence, I would opt for explicitly passing each prop if it is used in the `style` object

Reply:
Sure, then I will pass the props that are being used, which is much needed to control the bloat for the style instead of receiving the whole ``props``, but here is a doubt

For example if the component is using 3-4 attributes from props, then we might end up doing something like this

```<Container height={props.height} width={props.width} />```

What is your opinion we will choose to go with explicit now ?

Reply:
let's pass { ...props } if at least two values are used in the style object. and let's document this as well in the code with a simple explanation comment

Reply:
Done I will make the changes, if at least two values are being used then I will pass down the props or i will just send the value that is being used 👍